sdk/lib/core/pattern.dart

```diff
@@ -41,10 +41,10 @@ abstract class Pattern {
 /// The following example finds all matches of a [RegExp] in a [String]
 /// and iterates through the returned iterable of `Match` objects.
 /// ```dart
-/// RegExp exp = RegExp(r"(\w+)");
-/// String str = "Parse my string";
-/// Iterable<Match> matches = exp.allMatches(str);
-/// for (Match m in matches) {
+/// final regExp = RegExp(r"(\w+)");
+/// const string = 'Parse my string';
+/// final matches = regExp.allMatches(string);
+/// for (final Match m in matches) {
```

Comment:
Small improvements:  use ' instead of " in pattern (r'(\w+)' 
no need to define type for item in for-loop. 
_String match_, can be **var** or **final** match 


final regExp = RegExp(r'(\w+)');
const string = 'Parse my string';
final matches = regExp.allMatches(string);
for (final m in matches) {
  final match = m[0]!;
  print(match);
}

Reply:
fixed